source/kraken/realtime.cpp

```diff
@@ -492,10 +522,16 @@ create_disruption(const std::string& id,
                 impact->aux_info.stop_times.emplace_back(std::move(st_update));
             }
 
-            trip_effect = get_calculated_trip_effect(most_important_stoptime_status);
+            if (trip_update.trip().schedule_relationship() == trt::TripDescriptor_ScheduleRelationship_ADDED) {
+                // Yeah, that's quite hardcoded...
```

Comment:
:smile: we may add a mapping method to cover those.
Maybe use/adapt the `get_calculated_trip_effect` from below?

Reply:
We can't use this method but we can rename it like `get_calculated_stop_trip_effect`

Reply:
I prefer to keep this method only to find the effect inside stop time if it is needed

Reply:
As `get_calculated_trip_effect` will be removed soon, we may leave as it is